apps/astarte_trigger_engine/lib/astarte_trigger_engine/xandra_nodes.ex

```diff
@@ -0,0 +1,42 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.TriggerEngine.XandraNodes do
+  use Skogsra.Type
+
+  @impl Skogsra.Type
+  def cast(value)
+
+  def cast(value) when is_binary(value) do
+    value
+    |> String.split(",", trim: true)
+    |> Enum.map(fn host_port_str ->
+      case String.split(host_port_str, ":") do
+        [host, port] ->
+          Enum.join([host, port], ":")
+
+        [host] ->
+          Enum.join([host, "9042"], ":")
+      end
+    end)
```

Comment:
Actually, the default port in Xandra is already handled internally, `String.split`  will suffice (as it was before, see the deleted `:xandra_nodes` function in the `Config` module).
You can move this (with the appropriate transformation from `Enum.join` to a tuple creation) to CQExNodes, this correctly handles strings like `"localhost"`